saidump/saidump.cpp

```diff
@@ -399,7 +418,159 @@ void dumpGraph(const TableDump& td)
+*/
+int pre_process_file(string file_name)
+{
+    ifstream input_file(file_name);
+
+    if (!input_file.is_open())
+    {
+        cerr << "Failed to open the input file." << endl;
+        return SAI_STATUS_FAILURE;
+    }
+
+    input_file.seekg(0, ios::end);     // Move to the end of the file
+    int64_t file_size = input_file.tellg(); // Get the current position
+
+    if (file_size >= RDB_FILE_MAX_SIZE)
+    {
+        SWSS_LOG_ERROR("Get %s's size failure or its size %ld  >= %ld MB.\n", file_name.c_str(), file_size, RDB_FILE_MAX_SIZE / 1024 / 1024);
+
+        // Close the input file
+        input_file.close();
+        return SAI_STATUS_FAILURE;
+    }
+    input_file.seekg(0, ios::beg);     // Move to the begin of the file
+
+    // Read the content of the input file into a string
+    string content((istreambuf_iterator<char>(input_file)),
+                   istreambuf_iterator<char>());
+
+
+    input_file.close();
+
+    content = regex_replace(content,  regex("\\},\\{\\r"), ",");
+
+    //erase the 1st and last char.
+    if (content.length() >= 2 && content[0] == '[' && content[content.length()-1] == ']')  {
+        content.erase(0, 1);
+        content.erase(content.length() - 1);
+    }
+
+    ofstream outputFile(file_name);
+
+    if (!outputFile.is_open())
+    {
+        cerr << "Failed to open the output file." << endl;
+        return SAI_STATUS_FAILURE;
+    }
+
+    outputFile << content;
+    outputFile.close();
+
+    return SAI_STATUS_SUCCESS;
+}
+
+int dump_from_redis_rdb_file(string file_name)
+{
+    ifstream input_file(file_name);
+    if (!input_file.is_open())
+    {
+        SWSS_LOG_NOTICE("The file %s does not exist for dumping from redis dump.rdb file.", file_name);
+        return SAI_STATUS_FAILURE;
+    }
+
+    input_file.seekg(0, ios::end);     // Move to the end of the file
+    int64_t file_size = input_file.tellg(); // Get the current position
+    if (file_size >= RDB_FILE_MAX_SIZE)
+    {
+        SWSS_LOG_ERROR("Get %s's size failure or its size %ld  >= %ld MB.\n", file_name.c_str(), file_size, RDB_FILE_MAX_SIZE / 1024 / 1024);
+
+        // Close the input file
+        input_file.close();
+        return SAI_STATUS_FAILURE;
+    }
+    input_file.seekg(0, ios::beg);     // Move to the begin of the file
+
+    try
+    {
+        // Parse the JSON data from the file (validation)
+        nlohmann::json jsonData;
+        input_file >> jsonData;
+        input_file.close();
+
+        SWSS_LOG_DEBUG("%s\n","JSON file is valid.\n");
+
+        for (json::iterator it = jsonData.begin(); it != jsonData.end(); ++it)
+        {
+            json jj_key = it.key();
+
+            string keystr = jj_key;
+            string item_name = keystr;
+            size_t pos = keystr.find_first_of(":");
+            if (pos != string::npos)
+            {
+                if("ASIC_STATE" != keystr.substr(0, pos))  // filter out non ASIC_STATE
```

Comment:
use ASIC_STATE from sairediscommon.h const

Reply:
Fix it.